src/controller/event/handler/internal/util.go

```diff
@@ -43,3 +44,21 @@ func autoScan(ctx context.Context, a *artifact.Artifact, tags ...string) error {
 		return scan.DefaultController.Scan(ctx, a, options...)
 	})(orm.SetTransactionOpNameToContext(ctx, "tx-auto-scan"))
 }
+
+func autoGenSBOM(ctx context.Context, a *artifact.Artifact) error {
+	proj, err := project.Ctl.Get(ctx, a.ProjectID)
+	if err != nil {
+		return err
+	}
+	if !proj.AutoSBOMGen() {
+		return nil
+	}
+	// transaction here to work with the image index
+	return orm.WithTransaction(func(ctx context.Context) error {
+		options := []scan.Option{}
+		// TODO: extract the sbom scan type to a constant
+		options = append(options, scan.WithScanType("sbom"))
```

Comment:
should we use a constant variable with the value of `sbom`? instead of using the string literal `sbom` everywhere

Reply:
maybe we need something like this:
* https://github.com/goharbor/harbor/pull/20244/files

search `ScanTypeSbom`